actors/ppo_atari_pz.py

```diff
@@ -0,0 +1,1085 @@
+
+    @classmethod
+    def load(
+        cls, model_id: int, version_number: int, model_user_data: dict, version_user_data: dict, model_data_f: str
+    ) -> Model:
+        """Load the model"""
+        model = PPOModel(
+            model_id=model_id,
+            version_number=version_number,
+            environment_implementation=model_user_data["environment_implementation"],
+            num_actions=int(model_user_data["num_actions"]),
+            input_shape=literal_eval(model_user_data["input_shape"]),
+            num_policy_outputs=int(model_user_data["num_policy_outputs"]),
+            device=model_user_data["device"],
+        )
+
+        # Load the model parameters
+        network_state_dict = torch.load(model_data_f)
+        model.network.load_state_dict(network_state_dict)
+
+        # Load version data
+        model.iter_idx = version_user_data["iter_idx"]
+        model.total_samples = version_user_data["total_samples"]
+        return model
+
+
+class PPOActor:
+    """PPO actor"""
+
+    def __init__(self, _cfg):
+        self._dtype = torch.float
+
+    def get_actor_classes(self):
+        """Get actor"""
+        return [PLAYER_ACTOR_CLASS]
+
+    async def impl(self, actor_session):
+        # Start a session
+        actor_session.start()
+        config = actor_session.config
+
+        # Get observation and action space
+        environment_specs = EnvironmentSpecs.deserialize(config.environment_specs)
+        observation_space = environment_specs.get_observation_space()
+        action_space = environment_specs.get_action_space(seed=config.seed)
+
+        # Get model
+        model, _, version_info = await actor_session.model_registry.retrieve_version(
+            PPOModel, config.model_id, config.model_version
+        )
+        # log.info(f"Actor - retreved model number: {version_info['version_number']}")
+        obs_shape = model.input_shape[::-1]
+
+        async for event in actor_session.all_events():
+            if event.observation and event.type == cogment.EventType.ACTIVE:
+                if (
+                    event.observation.observation.HasField("current_player")
+                    and event.observation.observation.current_player != actor_session.name
+                ):
+                    # Not the turn of the agent
+                    actor_session.do_action(action_space.serialize(action_space.create()))
+                    continue
+                obs = observation_space.deserialize(event.observation.observation)
+                obs_tensor = torch.tensor(obs.flat_value, dtype=self._dtype).reshape(obs_shape)
+                obs_tensor = torch.unsqueeze(obs_tensor.permute((2, 0, 1)), dim=0).to(torch.device(model.device))
+
+                # Get action from policy network
+                with torch.no_grad():
+                    dist = model.network.get_action(obs_tensor)
+                    action = dist.sample().cpu().numpy()[0]
```

Comment:
Use a seeded RNG (call `torch.manual_seed(...)` before there creation of the distribution (perhaps as an argument of get_action)

Reply:
Done